sopel/config/core_section.py

```diff
@@ -96,8 +96,41 @@ class CoreSection(StaticSection):
     channels = ListAttribute('channels')
     """List of channels for the bot to join when it connects"""
 
+    db_type = ChoiceAttribute('db_type', choices=[
+        'sqlite', 'mysql', 'postgres', 'mssql', 'oracle', 'firebird', 'sybase'], default='sqlite')
+    """The type of database to use for Sopel's database.
+
+    mysql - pip install mysql-python (Python 2) or pip install mysqlclient (Python 3)
+    postgres - pip install psycopg2
+    mssql - pip install pymssql
```

Comment:
Why not add the required additional package(s) for each dialect in this `ChoiceAttribute` as [setuptools extras](https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-extras-optional-features-with-their-own-dependencies) (each extra named for its corresponding choice here)?

Can be done in a separate PR, though. Just wanted to put the bug in your ear!

Reply:
I'll throw this in a separate PR